genmod/score_variants/score_function.py

```diff
@@ -174,3 +174,45 @@ def set_equal(self):
         self.logger.debug("Setting equal to True")
         self._equal = True
         return
+
+    @property
+    def score_range(self):
+        if self._equal:
+            # Invalid request to expect a known range from an unknown plugin config
+            raise ValueError('User supplied score values does not have a known score range')
+
+        # Find out what map type is used
+        conf_value = True if len(self._value_dict) > 0 else False
+        conf_str = True if len(self._string_dict) > 0 else False
+        conf_tree = True if len(self._interval_tree) > 0 else False
+
+        if conf_value and not (conf_str or conf_tree):
```

Comment:
I think this if-else tree deserves a separate function, to explain the purpose snd make it possible tto unit test and bring the complexity of `def score_range(self)`